modin/engines/dask/pandas_on_dask_delayed/axis_partition.py

```diff
@@ -32,27 +32,23 @@ def apply(self, func, num_splits=None, other_axis_partition=None, **kwargs):
 
         if other_axis_partition is not None:
             return [
-                DaskRemotePartition(dask.delayed(obj))
-                for obj in deploy_func_between_two_axis_partitions(
+                DaskRemotePartition(obj)
+                for obj in dask.delayed(
+                    deploy_func_between_two_axis_partitions, nout=num_splits
+                )(
```

Comment:
Just to make sure, you should know that there won't be any parallelism here.  Every delayed call happens in a single task, so even if `num_splits` is high you're still just calling one function here in one thread.

Reply:
Yes, this abstraction is for bringing multiple smaller partitions together and operating on them as a whole, then re-splitting them. I do not expect parallelism here.